### PR TITLE
[types/react-native] Bump to 0.72

### PR DESCRIPTION
**Pull request by lunaleaps**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [ ] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: [react-native](https://github.com/facebook/react-native/tree/0.72-stable/packages/react-native/types)
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

### CODE REVIEW

**Comment by DangerBotOSS** [comment]
Inspecting the JavaScript source for this package found some properties that are not in the .d.ts files.
The check for missing properties isn't always right, so take this list as advice, not a requirement.
## react-native-material-kit ([<kbd>unpkg</kbd>](https://unpkg.com/browse/react-native-material-kit@latest/))
was missing the following properties:
1. mdl
1. mdl
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 4182f46fba4fcce38594c8cc6f5547dfbb6a27ad
</p>

File: types/react-native/Libraries/Components/TextInput/InputAccessoryView.d.ts [hunk 1]
```diff
@@ -14,7 +14,7 @@ export class InputAccessoryView extends React.Component<InputAccessoryViewProps>
 export interface InputAccessoryViewProps {
   backgroundColor?: ColorValue | undefined;
 
-  children?: React.ReactNode;
+  children?: React.ReactNode | undefined;
```
Comment (eps1lon) [inline types/react-native/Libraries/Components/TextInput/InputAccessoryView.d.ts:17]: `undefined` is already in `ReactNode` so this is redundant.

File: types/react-native/Libraries/Lists/SectionList.d.ts [hunk 2]
```diff
@@ -52,54 +52,12 @@ export type SectionListRenderItem<ItemT, SectionT = DefaultSectionT> = (
...
-   */
-  ListHeaderComponentStyle?: StyleProp<ViewStyle> | undefined | null;
-
   /**
    * Rendered in between each section.
    */
   SectionSeparatorComponent?:
     | React.ComponentType<any>
-    | React.ReactElement<unknown>
+    | React.ReactElement
```
Comment (eps1lon) [inline types/react-native/Libraries/Lists/SectionList.d.ts:60]: This should probably stay to not break assignability after `isValidElement` narrowing. See https://github.com/DefinitelyTyped/DefinitelyTyped/pull/64840

**Review by eps1lon — APPROVED** [review]

File: types/react-native/index.d.ts [hunk 3]
```diff
@@ -25,6 +25,7 @@
 //                 Xianming Zhong <https://github.com/chinesedfan>
 //                 Valentyn Tolochko <https://github.com/vtolochk>
 //                 Sergey Sychev <https://github.com/SychevSP>
+//                 Kelvin Chu <https://github.com/RageBill>
```
Comment (gabritto) [inline types/react-native/index.d.ts:28]: Why are you adding this owner?

Comment (gabritto) [inline types/react-native/index.d.ts:28]: Why are you adding this owner?

**Review by gabritto — CHANGES_REQUESTED** [review]
Please address @eps1lon's comments and the one I left, and then we can merge this. Thanks.

File: types/react-native/index.d.ts (hunk 3 above)
Comment (lunaleaps) [reply types/react-native/index.d.ts:28]: This was a copy from `react-native/types/index.d.ts` -- I guess this person added themselves to that one? Will remove

**Comment by sandersn** [comment]
Since the most recent commit addresses the review comments, I'm going to merge this.

**Review by sandersn — APPROVED** [review]

File: types/react-native/Libraries/Components/TextInput/InputAccessoryView.d.ts (hunk 1 above)
Comment (NickGerleman) [reply types/react-native/Libraries/Components/TextInput/InputAccessoryView.d.ts]: I think this was done by an ESLint plug-in ensuring compatibility with “ exactOptionalPropertyTypes”. The backing Flow types do not have a similar concept, so for consistency we enforce optional properties always can accept undefined values.

I think it’s just a simple AST plug-in that doesn’t follow the backing types.

**Pull request merged**